src/algebra/group/units.lean

```diff
@@ -33,6 +33,14 @@ structure add_units (α : Type u) [add_monoid α] :=
 
 attribute [to_additive add_units] units
 
+section has_elem
+
+@[to_additive, simp] lemma unique_has_one {α : Type*} [unique α] [has_one α] :
```

Comment:
I'm happy to merge the rest of this PR, but the various subtractions from calls to simp suggests this lemma is a bad idea.

Reply:
Let's see if adding the trivial ring lemma equating 1 to 0 helps.

Reply:
What do you think of the two new lemmas added?